### PR TITLE
Apple Silicon / M1 ARM improvements

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
A few quality of life improvements for Apple Silicon users:
* Support arm64 binaries when compiling and building the plugin package.
* Workaround https://github.com/npm/cli/issues/2846 with a `preinstall` and `postinstall` hook that removes any existing integrity checksum for `mattermost-webapp`, and ensures it doesn't get added again.

This PR also includes an `.nvmrc` (for all users) that pins Node at `16.4.0` when using the most excellent [Node Version Manager](https://github.com/nvm-sh/nvm).

#### Ticket Link
None.

#### Checklist
- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [ ] ~~Unit tests updated~~

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -70,11 +70,13 @@ ifneq ($(HAS_SERVER),)
 ifeq ($(MM_DEBUG),)
 	cd server && env GOOS=linux GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -o dist/plugin-linux-amd64;
 	cd server && env GOOS=darwin GOARCH=amd64 $(GO) build $(GO_BUILD_FLAGS) -o dist/plugin-darwin-amd64;
+	cd server && env GOOS=darwin GOARCH=arm64 $(GO) build $(GO_BUILD_FLAGS) -o dist/plugin-darwin-arm64;
```
Comment (cpoile) [inline Makefile:73]: I love how easy this is with Go. Very impressed.

**Review by cpoile — APPROVED** [review]
🙌

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+16.4.0
```
Comment (calebroseland) [inline .nvmrc:1]: Intentionally 16.4? Any reason we couldn't bump it to 16.13?

There's also https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines if we wanted to get more specific with compatibility markers.

**Review by calebroseland — APPROVED** [review]
Nice workaround!

**Pull request merged**